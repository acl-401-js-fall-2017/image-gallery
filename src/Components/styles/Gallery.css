.Gallery {
    height: 100%;
}

.imgSlide {
    height: 60%;
    width: 80%;
    margin: 1em auto;
    overflow: hidden;
}

.Gallery img {
    height: 90%;

}

.galleryButtons {
    display: flex;
    justify-content: center;
    height: 10%;
}

.galleryButtons h1 {
    width:50%;
    text-align: center;
    background-color: black;
    border-radius: 0.25em;
    color: white;
    margin: 0.75em 1em;
    line-height: 1.5em;
}