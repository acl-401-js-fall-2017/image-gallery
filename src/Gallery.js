import React, { PureComponent } from 'react';
import styled from 'styled-components';

export default class Thumbnail extends PureComponent {
  constructor() {
    super();
    this.state= { index: 0}
    this.history =[];
  }

  setState(state, ignore) {
    if(!ignore) this.history.push(state);
    super.setState(state);
  }

  handleClick = (value) =>{
    console.log('clicked');
    const newState ={
      ...this.state,
      index: this.state.index + value
    }
    this.setState(newState);
  }
  render() {
    const {gallery, shouldDisplay } =this.props
    return( 
      <StyledDiv shouldDisplay={shouldDisplay}>
        <div>
          {this.props.gallery.map((img, i) => (
            <ImgDiv shouldDisplay ={this.state.index === i}>
              <span> {img.title} </span>
              <img style={{ width:'100%' }} key={img._id} src={img.img} alt='image'/>
              <span> {img.description} </span>
              { i !== gallery.length -1 && <span onClick ={()=> this.handleClick(1)}> next </span>}
              {i !== 0 && <span onClick ={()=> this.handleClick(-1)}> previous</span>}
            </ImgDiv>
          ))}
        </div>
      </StyledDiv>
    )
  };
}

const StyledDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
`;

const ImgDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
flex-direction: column;
margin: 0 10%;
`;

