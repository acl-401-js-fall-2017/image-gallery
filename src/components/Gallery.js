import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Bunnies from '../images/bunnies';



export default class Gallery extends PureComponent {

    static propTypes = {
      bunnies: PropTypes.arrayOf(PropTypes.object).isRequired
    }

    constructor(){
      super();
      this.state = {
        bunnies: Bunnies,
        picIndex: 0
      };
    }


    nextImage = ({ target }) => {
      this.setState(prevState => {
        let picIndex;
        if (target.id === 'prevButton') {
          picIndex = prevState.picIndex - 1;
        }
        if (target.id === 'nextButton') {
          picIndex = prevState.picIndex + 1;
        }
        return { picIndex };
      });
    }

    render() {
      const bunny = this.props.bunnies[this.state.picIndex];

      return (
        <div className="gallery">
          <img src={bunny.url} alt={bunny.title} height="100px"/>
          <h4>{bunny.title}</h4>
          <h6>{bunny.description}</h6>

          {this.state.picIndex !== 0 && (
            <button type="button" value="Previous" id="prevButton" onClick={this.nextImage}>
          Previous
            </button>
          )}

          {this.state.picIndex !== this.state.bunnies.length - 1  && (
            <button type="button" value="Next" id="nextButton" onClick={this.nextImage}>
          Next
            </button>
          )}
          {/* Might want to add remove button here */}
        </div>
      );
    }
}