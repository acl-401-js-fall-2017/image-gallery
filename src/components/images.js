import React, { Component } from 'react';
import '../App.css';
import shortid from 'shortid';
import WonderList from '../views/list';
import WonderThumbnail from '../views/thumbnail';
import WonderGallery from '../views/gallery';
import AddImage from '../AddImage';
import { createWonder } from '../actions';
import {
  Route,
  Switch
} from 'react-router-dom';


class Images extends Component {
  constructor(){
    super();
    this.state = {
      view: 'list',
      wonders: [
        {
          _id: shortid.generate(),
          title: 'Great Pyramid of Giza',
          description : 'The Great Pyramid of Giza is the oldest and largest of the three pyramids in the Giza pyramid complex bordering what is now El Giza, Egypt. It is the oldest of the Seven Wonders of the Ancient World, and the only one to remain largely intact.',
          url:'https://upload.wikimedia.org/wikipedia/commons/e/e3/Kheops-Pyramid.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'Great Wall of China',
          description : 'The Great Wall of China is a series of fortifications made of stone, brick, tamped earth, wood, and other materials, generally built along an east-to-west line across the historical northern borders of China to protect the Chinese states and empires against the raids and invasions of the various nomadic groups of the Eurasian Steppe.',
          url:'https://upload.wikimedia.org/wikipedia/commons/6/6f/The_Great_Wall_of_China_at_Jinshanling.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'Petra',
          description : 'Petra, originally known to the Nabataeans as Raqmu, is a historical and archaeological city in southern Jordan. The city is famous for its rock-cut architecture and water conduit system. Another name for Petra is the Rose City due to the color of the stone out of which it is carved.',
          url:'https://upload.wikimedia.org/wikipedia/commons/b/b7/The_Monastery%2C_Petra%2C_Jordan8.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'The Colosseum',
          description : 'The Colosseum or Coliseum , also known as the Flavian Amphitheatre (Latin: Amphitheatrum Flavium; Italian: Anfiteatro Flavio [aŋfiteˈaːtro ˈflaːvjo] or Colosseo [kolosˈsɛːo]), is an oval amphitheatre in the centre of the city of Rome, Italy. Built of concrete and stone,[1] it is the largest amphitheatre ever built.',
          url:'https://upload.wikimedia.org/wikipedia/commons/f/f7/KOLOSSEUM_-_panoramio.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'Chichen Itza',
          description : 'Chichen Itza was one of the largest Maya cities, with the relatively densely clustered architecture of the site core covering an area of at least 5 square kilometres (1.9 sq mi). Smaller scale residential architecture extends for an unknown distance beyond this. The city was built upon broken terrain, which was artificially levelled in order to build the major architectural groups, with the greatest effort being expended in the levelling of the areas for the Castillo pyramid, and the Las Monjas, Osario and Main Southwest groups.',
          url:'https://upload.wikimedia.org/wikipedia/commons/7/7a/Chichen-Itza-Castillo-Seen-From-East.JPG'
        },
        {
          _id: shortid.generate(),
          title: 'Machu Picchu',
          description : 'Most archaeologists believe that Machu Picchu was constructed as an estate for the Inca emperor Pachacuti (1438–1472). Often mistakenly referred to as the "Lost City of the Incas" (a title more accurately applied to Vilcabamba), it is the most familiar icon of Inca civilization. The Incas built the estate around 1450 but abandoned it a century later at the time of the Spanish Conquest',
          url:'https://upload.wikimedia.org/wikipedia/commons/e/eb/Machu_Picchu%2C_Peru.jpg'
        },
        {
          _id: shortid.generate(),
          title: 'Taj Mahal',
          description : 'The Taj Mahal is an ivory-white marble mausoleum on the south bank of the Yamuna river in the Indian city of Agra. It was commissioned in 1632 by the Mughal emperor, Shah Jahan (reigned 1628–1658), to house the tomb of his favourite wife, Mumtaz Mahal. The tomb is the centrepiece of a 17-hectare (42-acre) complex, which includes a mosque and a guest house, and is set in formal gardens bounded on three sides by a crenellated wall.',
          url:'https://upload.wikimedia.org/wikipedia/commons/c/c8/Taj_Mahal_in_March_2004.jpg'
        }
      ]
    };
    this.handleOnSubmitCreateWonder = this.handleOnSubmitCreateWonder.bind(this);
  }

 
  handleOnSubmitCreateWonder = (title, description, url) => {
    const newState = createWonder(this.state, title, description, url);
    this.setState(newState);
  }
  

  render() {
    const { wonders } = this.state;
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Wonders of The World!!!</h1>
        </header>
        <div>
          <Switch>
            <Route exact path={`${this.props.match.url}/list`} render={() => <WonderList wonders={wonders}/>}/>
            <Route exact path={`${this.props.match.url}/list`} render={() => <WonderThumbnail wonders={wonders}/>}/>
            <Route exact path={`${this.props.match.url}/:gallery?`}  render={() => <WonderGallery wonders={wonders}/>}/>
          </Switch>
        </div>
        <AddImage handleOnSubmit={this.handleOnSubmitCreateWonder}/>
      </div>
    );
  }
}

export default Images;
