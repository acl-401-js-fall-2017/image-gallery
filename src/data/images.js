const imageData = [
  {
    title: 'Culprit Apprehended',
    description: 'Milk was left home alone.',
    url: '/images/20160310_214017.jpg'
  },
  {
    title: 'Sunset on Marine Drive',
    description: 'Milk joined us for some longboarding by Marine Dr.',
    url: '/images/20160416_194337.jpg'
  },
  {
    title: 'First Meal Together',
    description: 'Oreo and Milk sharing their first meal together after a tumultuous first meeting.',
    url: '/images/20160527_132115.jpg'
  }
];

export default imageData;