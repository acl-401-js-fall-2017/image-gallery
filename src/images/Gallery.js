import React, { PureComponent } from 'react';
import styled from 'styled-components';
import { loadGallery, deleteImg } from '../utils/actions';
import imgApi from '../utils/services/imgApi';


export default class Gallery extends PureComponent {
  state = { 
    index: 0, 
    gallery:[]
  }

  async componentDidMount() {
    const img = await imgApi.get();
    const newState = loadGallery(this.state, img);
    this.setState(newState);
  }

  handleDelete = async id => {
    await imgApi.remove(id);
    const newState = deleteImg(this.state, id);
    this.setState(newState);
  }


  handleClick = (value) => {
    console.log('clicked');
    const newState = {
      ...this.state,
      index: this.state.index + value
    };
    this.setState(newState);
  }

  render() {
    return( 
      <StyledDiv>
        <div>
          {this.state.gallery.map((img, i) => (
            <ImgDiv key={img._id} shouldDisplay ={this.state.index === i}>
              <span> {img.title} </span>
              <DeleteDiv onClick={() => this.handleDelete(img._id)}>X</DeleteDiv>
              <img style={{ width:'100%' }} src={img.url} alt=''/>
              <span> {img.description} </span>
              { i !== this.state.gallery.length -1 && <span onClick ={()=> this.handleClick(1)}> next </span>}
              {i !== 0 && <span onClick ={()=> this.handleClick(-1)}> previous</span>}
            </ImgDiv>
          ))}
        </div>
      </StyledDiv>
    );
  }
}

const StyledDiv = styled.div`
display: 'flex';
`;

const ImgDiv = styled.div`
display:${props => props.shouldDisplay ? 'flex' : 'none'};
flex-direction: column;
margin: 0 10%;
`;

const DeleteDiv = styled.div`
margin-left: 90%;
text-align: center;
border: 1px solid black;
`;