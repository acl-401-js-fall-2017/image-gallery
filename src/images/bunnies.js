//import shortid from 'shortid';

const Bunnies = [
  {
    _id: '1',
    title: 'Best Bunny',
    description: 'Just the cutest color on a bunny ever',
    url: 'https://s3.amazonaws.com/f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
  },
  {
    _id: '2',
    title: 'Kind of cute Bunny',
    description: 'Too many carbs :(',
    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg'
  },
  {
    _id: '3',
    title: 'Tiny Bunny',
    description: 'Cant see me',
    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg'
  }
];

export default Bunnies;