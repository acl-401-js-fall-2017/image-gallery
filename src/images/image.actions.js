import shortid from 'shortid';

export function createImage(state, title, description, url) {
  const newImage = { _id: shortid.generate(), title, description, url };
  return {
    ...state,
    images: [
      ...state.images,
      newImage
    ]
  };
}


export function addImage(state, image) {
  return {
    ...state,
    images: [
      ...state.images,
      image
    ]
  };
}

export function removeImage(state, _id) {
  const index = state.images.findIndex(item => item._id === _id);
  if(index === -1) return state;

  const images = state.images.slice();
  images.splice(index, 1);
  
  return {
    ...state,
    images
  };
}

export function updateImage(state, image) {
  const { images } = state;
  const index = images.findIndex(item => item._id === image._id);
  if(index === -1) return state;
  
  return {
    ...state,
    images: [
      ...images.slice(0, index),
      image,
      ...images.slice(index + 1)
    ]
  };
}