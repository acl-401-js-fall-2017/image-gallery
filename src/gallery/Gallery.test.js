import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import images from '../images';
import Gallery from './Gallery';

describe('List', () => {
  it('shows gallery', () => {
    const wrapper = shallow(<Gallery images={images}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});