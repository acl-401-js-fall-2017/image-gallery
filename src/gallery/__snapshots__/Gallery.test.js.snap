// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List shows gallery 1`] = `
<div
  className="gallery"
>
  <img
    alt="Cute Bunny"
    height="300px"
    src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
  />
  <h3>
    Cute Bunny
  </h3>
  <h6>
    Isn't it fuzzy-wuzzy cutest thing you've ever seen?
  </h6>
  <button
    id="prevButton"
    onClick={[Function]}
    type="button"
    value="Previous"
  >
    Previous
  </button>
  <button
    id="nextButton"
    onClick={[Function]}
    type="button"
    value="Next"
  >
    Next
  </button>
</div>
`;
